### PR TITLE
Home redesign

### LINKED ISSUE #37: Home page design update
Update the home page to match new v3 designs

### PR DESCRIPTION
**Pull request by tunangst**
# Description

Please include a brief summary of what the issue is and what has been done to solve it?

The issues: This started as an older design that had a few modules that were not needed anymore and a few that needed to be added.
To solve this issue I removed the components that were not needed and created my own. 
**It's time to change module:** created grid layout, added background image, put gradient over background image. The arch was created with an ::after element.
**Take action module:** minor adjustments, arrow had to be adjusted.
**Active campaigns:** made a new module for the cards *this may benefit for a component since I saw this on a different page*. Elastic grid layout housing the campaign cards. Cards have a text-shadow as well as a gradient over the image. ::hover with a blackout fade in with text.
**Wave Break module:** I added an SVG for this wave shape, I thought it would be easiest to do it this way and can be reused if necessary.
**How it works module:** Minimized the containers, adjusted minor styling, set up grids. Added the slanted background with ::before and ::after shapes, I thought it was easier and efficient this way.
**Partners:** added a card component for the partners for consistency and reusability. There is a hover effect that may not be what the designer wanted, but that can be up for review.
**Newsletter:** Added module, 2 backgrounds one from a ::before to get perfectly in the center. I added some code for basic validation which could be up for review as well. The inserted text in email section will get trim()ed of spaces and stored in state. Checkmarks are just CSS and ::before with checkmark content. Email form action goes to the same place as the email modal.

All of these modules I tried to simplify the massive amounts of divs, also tried to add semantic html to make things clear.

Fixes #37 

# Checklist:

- [✔] I have commented my code, particularly in hard-to-understand areas
- [✔] I have reviewed the 'files changed' tab on github to ensure all changes are expected
- [✔] I have added someone to review the pr

comments:
I believe my code is understandable without comments. Caught some SASS mistakes in files changed.

### CODE REVIEW

File: src/domain/home/Homepage.jsx [hunk 1]
```diff
@@ -1,27 +1,47 @@
...
-    const fetchCampaigns = async () => {
-      const resp = await fetch('https://api.now-u.com/api/v1/campaigns');
-      const campaigns = await resp.json();
-      setCampaigns(campaigns?.data);
-    };
-    fetchCampaigns();
-  }, []);
+    useEffect(() => {
+        const fetchCampaigns = async () => {
+            const resp = await fetch('https://api.now-u.com/api/v1/campaigns');
```
Comment (shah-tejas) [inline src/domain/home/Homepage.jsx:18]: Using the `campaignsURL` defined in [`constants.js`](https://github.com/now-u/now-u-campaigns/blob/1f17ad8c908c7c4b2c3c3833add174c002dc9c9c/src/utils/constants.js#L25) would be better. If the API URL changes, we can just change in the constants file:

File: src/domain/home/Homepage.jsx [hunk 2]
```diff
@@ -1,27 +1,47 @@
...
-    <div className={classes.homepageContainer}>
-      <Intro />
-      <HowItWorks />
-      <ActiveCampaigns campaigns={campaigns} />
-      <TheApp />
-    </div>
-  );
+        const fetchPartners = async () => {
+            const respPartners = await fetch(
+                'https://api.now-u.com/api/v1/organisations'
```
Comment (shah-tejas) [inline src/domain/home/Homepage.jsx:26]: Similar to `campaignsURL`, `organisationsURL` can be created.

File: src/domain/home/components/partners/Partners.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                <Text type='h1' className={classes.header}>
+                    Our Partners
+                </Text>
+                <Text type='p' className={classes.description}>
+                    We work with many amazing charities and social enterprises.
+                    Want to find out more about partnering with now-u? Download
+                    our{' '}
+                    <a
+                        href={
+                            'https://share.now-u.com/partners/now-u_charity_collaboration_&_partnership_guide.pdf'
```
Comment (shah-tejas) [inline src/domain/home/components/partners/Partners.jsx:20]: Would be nice to add this URL to `constants.js` as well.

File: src/domain/home/Homepage.jsx (hunk 2 above)
Comment (tunangst) [reply src/domain/home/Homepage.jsx]: good catch

**Comment by tunangst** [comment]
> Looks pretty good to me! The mixin looks very cool, and fits perfectly!
> I feel some of the URL's could be mentioned in the constants, making it easy to update in the future.
> 
> Also, I noticed the Subscribe button acting weird:
> 
> Screen Shot 2020-08-19 at 6 55 54 PM

Thank you, this brought up a major situation that I was not aware of. fieldsets can't be flexed in chrome. Would you mind retesting to see if this bug is fixed on your end too?

File: src/domain/home/components/active_campaigns/campaign_card/CampaignCard.module.scss [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            color: $white;
+            font-size: 30px;
+            margin: 0;
+            filter: drop-shadow(1px 1px 3px rgba($black, 0.8));
+        }
+    }
+    &::before {
+        z-index: 1;
+        opacity: 0;
+        content: 'View Campaign';
```
Comment (JElgar) [inline src/domain/home/components/active_campaigns/campaign_card/CampaignCard.module.scss:50]: Note: I think we are trying to be consistent with the style of Captial first word and not capitalising the rest. (We aren't doing very well tho so there are probably many cases of this throughout the app and website)
```suggestion
        content: 'View campaign';
```

**Review by JElgar — APPROVED** [review]
Smashed it! Looks awesome. Only 2 things Id say is:

1. I completely agree with @shah-tejas can we get those urls into `constants.js`. 
2. A tiny thing - I think we are going for Captial first word and lower case rest eg "View campaign"

**Comment by JElgar** [comment]
> Thank you, this brought up a major situation that I was not aware of. fieldsets can't be flexed in chrome. Would you mind retesting to see if this bug is fixed on your end too?
LGTM

**Comment by JElgar** [comment]
Also one more thing! Can the partner tiles link to the partner's website (this should Fix #82). They should be part of the response from the api already.

**Comment by JElgar** [comment]
Love it, that looks great to me

**Pull request merged**